appstudio-controller/controllers/appstudio.redhat.com/environment_controller.go

```diff
@@ -385,48 +424,52 @@ func generateDesiredResource(ctx context.Context, env appstudioshared.Environmen
 				"DeploymentTarget not found for DeploymentTargetClaim", &env,
 				EnvironmentConditionErrorOccurred, metav1.ConditionTrue, EnvironmentReasonErrorOccurred, log); err != nil {
 
-				return nil, true, fmt.Errorf("unable to update environment status condition. %v", err)
+				return nil, errorConditionSet_false, fmt.Errorf("unable to update environment status condition. %v", err)
 			}
 
-			return nil, true, nil
+			return nil, errorConditionSet_true, nil
 		}
 
-		// Update Status.Conditions field of Environment as false if error is resolved
-		if err = updateConditionErrorAsResolved(ctx, k8sClient, "", &env, EnvironmentConditionErrorOccurred, metav1.ConditionFalse, EnvironmentReasonErrorOccurred, log); err != nil {
-			return nil, true, err
+		var namespacesField []string
+
+		if dt.Spec.KubernetesClusterCredentials.DefaultNamespace != "" {
```

Comment:
BUG FIX: We were missing adding the `DefaultNamespace` to the ManagedEnvironment `namespaces` field.